include/TFEL/Math/Stensor/Internals/HarariEigenSolver.hxx

```diff
@@ -0,0 +1,140 @@
+/*!
+ * \file   HarariEigenSolver.hxx
+ * \brief
+ * \author Maxence Wangermez/Thomas Helfer
+ * \date   13/02/2024
+ * \copyright Copyright (C) 2006-2024 CEA/DEN, EDF R&D. All rights
+ * reserved.
+ *
+ * This project is publicly released under either the GNU GPL Licence
+ * or the CECILL-A licence. A copy of thoses licences are delivered
+ * with the sources of TFEL. CEA or EDF may also distribute this
+ * project under specific licensing conditions.
+ *
+ * This file was adapted from Numerical diagonalization of 3x3
+ * matrices of Isaac Harari and Uri Albocher, see
+ * [1] Computation of eigenvalues of a real, symmetric 3 × 3
+ * matrix with particular reference to the pernicious
+ * case of two nearly equal eigenvalues
+ *
+ */
+
+#ifndef LIB_TFEL_MATH_HARARIEIGENSOLVER_HXX
+#define LIB_TFEL_MATH_HARARIEIGENSOLVER_HXX
+
+#include <iostream>
+#include <cmath>
+#include <vector>
+#include "TFEL/Math/tvector.hxx"
+#include "TFEL/Math/tmatrix.hxx"
+
+namespace tfel::math::internals {
+
+  template <typename t>
+  int sign(t val);
+
+  template <typename T, typename A>
+  int argmax(std::vector<T, A> const&);
```

Comment:
I am quite surprise about `std::vector` usage. `std::vector` implies heap allocation, this shall be prohibited.

It seems that this function is unused, so we could remove it.